pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -132,6 +132,7 @@ func ValidateUsageOfDeclaredParameters(ctx context.Context, steps []Step, params
 	var errs *apis.FieldError
 	_, _, objectParams := params.SortByType()
 	allParameterNames := sets.NewString(params.GetNames()...)
+	allParameterNames.Insert(ReservedParamName)
```

Comment:
May be here we should check that the `ReservedParamName` is not already in the set. If it is then a user also declared it and it should throw an error since it is reserved?

Reply:
this check is not done from our side, but we're sure that users cannot declare the param with reserved name. :D

Reply:
(and we cannot do it from out side for some reason...